src/app/common/account-restoration/account-restore.ts

```diff
@@ -0,0 +1,67 @@
+import { createCounter } from '../utils/counter';
+import { fibonacciGenerator } from '../utils/fibonacci';
+
+const numOfEmptyAccountsToCheck = 20;
+
+interface AccountIndexActivityCheckHistory {
+  index: number;
+  hasActivity: boolean;
+}
+
+function minNumberOfAccountsNotChecked(num: number) {
+  return num < numOfEmptyAccountsToCheck;
+}
+
+function anyOfLastCheckedAccountsHaveActivity(arr: AccountIndexActivityCheckHistory[]) {
+  return arr.slice(arr.length - numOfEmptyAccountsToCheck).some(ob => ob.hasActivity);
+}
+
+function returnHighestIndex(arr: AccountIndexActivityCheckHistory[]) {
+  return Math.max(0, ...arr.filter(ob => ob.hasActivity).map(ob => ob.index));
+}
+
+async function recurseUntilGeneratorDone(generator: AsyncGenerator): Promise<any> {
+  const result = await generator.next();
+  if (result.done) return result.value;
+  return recurseUntilGeneratorDone(generator);
+}
+
+interface RecurseAccountsForActivityArgs {
+  doesAddressHaveActivityFn(index: number): Promise<boolean>;
+}
+export async function recurseAccountsForActivity({
```

Comment:
Would it be worthwhile to add some docs as to how this file was put together?

Reply:
Added a comment